include/aspect/time_stepping/repeat_on_cutback.h

```diff
@@ -0,0 +1,78 @@
+/*
+  Copyright (C) 2018 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef _aspect_time_stepping_repeat_on_cutback_h
+#define _aspect_time_stepping_repeat_on_cutback_h
+
+#include <aspect/time_stepping/interface.h>
+
+namespace aspect
+{
+  namespace TimeStepping
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements a time stepping plugin to repeat a time step if the
+     * next time step is significantly smaller than the last step.
+     *
+     * @ingroup TimeStepping
+     */
+    template <int dim>
+    class RepeatOnCutback : public Interface<dim>, public SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * Constructor.
+         */
+        RepeatOnCutback () = default;
+
+        /**
+         * @copydoc aspect::TimeStepping::Interface<dim>::execute()
+         */
+        virtual
+        double
+        execute() override;
+
+        /**
+         * The main execute() function.
+         */
+        virtual
+        std::pair<Reaction, double>
+        determine_reaction(const TimeStepInfo &info) override;
+
+        static
+        void
+        declare_parameters (ParameterHandler &prm);
+
+        virtual
+        void
+        parse_parameters (ParameterHandler &prm);
+
+      private:
+        double cut_back_amount;
+        double repeat_threshold;
```

Comment:
The members need some documentation, at least if they are upper bounds or prefactors.